exercises/practice/pig-latin/.meta/src/PigLatin.example.elm

```diff
@@ -0,0 +1,56 @@
+module PigLatin exposing (translate)
+
+
+translate : String -> String
+translate =
+    String.words >> List.map translateWord >> String.join " "
+
+
+translateWord : String -> String
+translateWord input =
+    if String.startsWith "xr" input || String.startsWith "yt" input then
+        input ++ "ay"
+
+    else if String.startsWith "y" input then
+        String.dropLeft 1 input ++ "yay"
+
+    else
+        let
+            ( consonnants, rest ) =
+                splitConsonant input
+        in
+        if String.endsWith "q" consonnants && String.startsWith "u" rest then
+            String.dropLeft 1 rest ++ consonnants ++ "uay"
+
+        else
+            rest ++ consonnants ++ "ay"
+
+
+splitConsonant : String -> ( String, String )
+splitConsonant input =
+    let
+        isVowelOrY char =
+            List.member char [ 'a', 'e', 'i', 'o', 'u', 'y' ]
+
+        ( first, second ) =
+            input |> String.toList |> takeWhile (isVowelOrY >> not)
+    in
+    ( String.fromList first, String.fromList second )
+
+
+takeWhile : (a -> Bool) -> List a -> ( List a, List a )
```

Comment:
This is picky, but I would probably rename this function. takeWhile doesn't seem to be what it is doing really. Maybe `split` or `slice` or something like that?

Reply:
It takes the first part of a list while a condition is true. The only difference with the usual `takeWhile` is that it returns both parts of the list, and that's clear with the type signature. I think the name makes sense.